### PR TITLE
Dispatch 'maybe_redirect_to_onboarding' action after onboarding redirection logic in `WC_Payments_Admin`

### LINKED ISSUE #2513: Newly onboarded account connection occasionally not being finalized
While testing a different PR, @ricardo and I noticed an issue where it sometimes takes you **back to the "Finish setup"** screen (instead of "Overview") immediately **after fully completing the onboarding** flow.

It appears that the hook [here](https://github.com/Automattic/woocommerce-payments/blob/aabf276642607ea26491dac05babadf29927b63a/includes/admin/class-wc-payments-admin.php#L68) is triggered prior to the hook [here](https://github.com/Automattic/woocommerce-payments/blob/aabf276642607ea26491dac05babadf29927b63a/includes/class-wc-payments-account.php#L43), causing the plugin to not recognize the onboarded account. It seems to happen _intermittently_, perhaps due to a race condition with the webhook handled [here](https://github.com/Automattic/woocommerce-payments/blob/aabf276642607ea26491dac05babadf29927b63a/includes/admin/class-wc-rest-payments-webhook-controller.php#L123-L125).

### PR DESCRIPTION
**Pull request by ricardo**
Fixes #2513

As described in the issue, there's probably a race condition somewhere which occasionally makes the onboarding redirection not finalize the account connection in the client.

When testing with a local server, the onboarding redirection will fail most times, while testing with the production server, it seems to _almost always_ work (p1626741938458500-slack-CGGCLBN58).

#### Changes proposed in this Pull Request

The `admin_menu` hook is triggered before `admin_init`, so this PR changes when `maybe_redirect_to_onboarding` is called in `includes/admin/class-wc-payments-admin.php`, to make sure it's called only after the onboarding redirection logic.

#### Testing instructions

Since this seems to be a race condition problem (which I didn't identify), and works intermittently, I'd ask you to try to reproduce the issue in `develop` and in this branch a couple times, so we can confirm a consistent behavior and see if this change is really what's fixing the issue:

1. Remove any existing account record from `wcpay_accounts`.
2. Try onboarding in dev mode with API requests being redirected to a local server.
3. After KYC completion, most of the time you should be redirected to `/payments/connect`. (I'd say when testing with a local server, 9 out of 10 times I'd be able to reproduce the issue).
4. Please repeat the steps above while in this branch and notice you should be correctly redirected to `/payments/overview` at all attempts.

-------------------

- [x] Added changelog entry (or does not apply)
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

File: includes/admin/class-wc-payments-admin.php [hunk 1]
```diff
@@ -65,7 +65,7 @@ public function __construct(
 
 		// Add menu items.
 		add_action( 'admin_menu', [ $this, 'add_payments_menu' ], 0 );
-		add_action( 'admin_menu', [ $this, 'maybe_redirect_to_onboarding' ], 1 );
+		add_action( 'admin_init', [ $this, 'maybe_redirect_to_onboarding' ], 11 ); // Run this after the WC setup wizard and onboarding redirection logic.
```
Comment (aprea) [inline includes/admin/class-wc-payments-admin.php:68]: With this change we might want to add the following code to the top of `maybe_redirect_to_onboarding` for added safety:

```php
if ( wp_doing_ajax() ) {
	return false;
}
```

**Comment by aprea** [comment]
I wasn't able to reproduce the issue on `develop` using the following steps:

- Delete all existing accounts from the `wcpay_accounts` table in `woocommerce-payments-server`.
- Clear account cache from WCPay Dev Utils
- Complete onboarding/Stripe KYC

I tried 3 times and each time I was successfully redirected to the overview screen after submitting the KYC.

I also tested on this branch and everything appears to be working as expected.

I've added one comment but otherwise this LGTM.

**Review by reykjalin — APPROVED** [review]
I was also unable to reproduce the issue in either `develop` or this branch, but the changes look good either way.

I think we can ship 🚀

**Comment by dechov** [comment]
I just tested in a different branch and got redirected (and the connection didn't get finalized / cached), and then tested in this branch and it worked fine. I'm testing locally without webhook forwarding (in case that matters).

File: includes/admin/class-wc-payments-admin.php (hunk 1 above)
Comment (dechov) [reply includes/admin/class-wc-payments-admin.php:68]: Sounds like a good idea.

Comment (naman03malhotra) [reply includes/admin/class-wc-payments-admin.php:68]: Included here: https://github.com/Automattic/woocommerce-payments/pull/2514/commits/1332331db725a5e931f51d5e8abc5664a73a1c75

**Review by dechov — APPROVED** [review]

**Pull request merged**